woocommerce/woocommerce-android
# [Google for Woo] Refactor Webview usage to use its own GoogleAdsWebView

Please do not merge until target is trunk.

This refactor is the implementation for the discussion in a previous PR:

https://github.com/woocommerce/woocommerce-android/pull/12013#discussion_r1683048918

Despite the big PR warning, most of this is just from the deletion of `ExitAwareWebView`.

### Description


We're adding a flag to tell `WPComWebView` to not authenticate if not needed, which is needed as a workaround for an issue related to Jetpack CP sites using Google for Woo features in the app.

The removes the need for having a separate `ExitAwareWebView`, which is now deleted.

Additionally, this PR also adds an event flow in `WPComWebView`. It will be needed later for Google for Woo tracking, so for now it just needs its code checked. The idea is to inject `SharedWebViewFlow` into any view model that needs it, and it can collect the `webViewEventFlow` to act on any event that it cares about ([example usage in the other PR](https://github.com/woocommerce/woocommerce-android/pull/12098/files#diff-1d2a6d541d2eb6081ac47608edf4914fb2b4292fff68a08e5ce8a1d24c9cb0a5R49))

### To test

Same testing instructions as in https://github.com/woocommerce/woocommerce-android/pull/11966

1. Test with a site on WordPress.com,
2. Use the Google for WooCommerce button in More Menu. The webview should automatically log you in and open the Google Ads page,
3. Switch to a site connected with application password/Jetpack CP,
4. Use the Google for WooCommerce button in More Menu. The webview should open a .org login form,
5. Login and ensure it opens the Google Ads page,
6. Close the webview and use the Google for WooCommerce button again,
7. Ensure that it opens the Google Ads page immediately, no login needed.
